observability/WFLY-17144-add-micrometer-support.adoc

```diff
@@ -0,0 +1,88 @@
+ = Add Micrometer Support to WildFly
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WildFly already supports basic metrics in the base WildFly system, as well as MicroProfile Metrics in the WildFly MicroProfile configuration. In the Java space, however, the library of choice, and the emerging de facto standard for Java developers is https://micrometer.io[Micrometer].  This RFE will add support for Micrometer to WildFly, providing an alternative to the WildFly Metrics subsystem, as well as providing a replacement for MicroProfile Metrics, which is slated for removal from WildFly under another JIRA.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17144
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1686 - Implement the Observability policy - Metrics (XP)
+* https://issues.redhat.com/browse/EAP7-1805 - Implement the Observability policy - Metrics - Base EAP
+* https://issues.redhat.com/browse/WFLY-17138 - Convert MP Metrics subsystem to Admin-Only Mode
+* https://issues.redhat.com/browse/EAP7-1975 - Remove MP Metrics functionality and convert subsystem to Admin-Only Mode
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [X] Engineering
+* [ ] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+* Metrics
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* The server/domain administrator should be able to configure Micrometer metrics
+* The extension/subsystem will not be added automatically to standard configs and must be installed manually. 
+** This is to prevent the dual gathering of metrics with WildFly Metrics still enabled. T
+** Admins are expected to pick one system or the other and modify local configs as required.
+* Server and JVM metrics tracked and exposed
+* Metrics will be exported via the OTLP protocol to a compatible collector via the default (localhost) endpoint.
```

Comment:
The [OTLP Meter registry implementation](https://github.com/micrometer-metrics/micrometer/blob/main/implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMeterRegistry.java#L58) is a push based one. What is the default `localhost` endpoint related to? Can you please elaborate on this detail to clarify how the export process works? 
I'd specifically add some explanation about the _collector_. Links to the local implementation could help understand the feature components topology and workflow. 

This is important, given that this registry type is not even documented upstream, see https://github.com/micrometer-metrics/micrometer-docs/issues/267

Reply:
I've expanded the text here.

Reply:
Looks great, thanks.